commands/permissions.go

```diff
@@ -109,26 +173,147 @@ func showRoleCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+				for _, val := range vals {
+					if val.Id == permissionID {
+						if _, ok := consolePermissionMap[key]; ok {
+							usedBy[key] = true
+						}
+					}
+				}
+			}
+			for key := range usedBy {
+				usedByIDs = append(usedByIDs, key)
+			}
+		}
+		return usedByIDs
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+
+	// Only show the 3-column view if the role has sysconsole permissions
+	// sysadmin has every permission, so no point in showing the "Used by"
+	// column.
+	if len(consolePermissionMap) > 0 && role.Name != "system_admin" {
+		fmt.Fprintf(w, "Property\tValue\tUsed by\n")
+		fmt.Fprintf(w, "--------\t-----\t-------\n")
+		fmt.Fprintf(w, "Name\t%s\t\n", role.Name)
+		fmt.Fprintf(w, "DisplayName\t%s\t\n", role.DisplayName)
+		fmt.Fprintf(w, "BuiltIn\t%v\t\n", role.BuiltIn)
+		fmt.Fprintf(w, "SchemeManaged\t%v\t\n", role.SchemeManaged)
+		for i, perm := range role.Permissions {
+			if i == 0 {
+				fmt.Fprintf(w, "Permissions\t%s\t%v\n", role.Permissions[0], strings.Join(getUsedBy(role.Permissions[0]), ", "))
+			} else {
+				fmt.Fprintf(w, "\t%s\t%v\n", perm, strings.Join(getUsedBy(perm), ", "))
+			}
+		}
+	} else {
+		fmt.Fprintf(w, "Property\tValue\n")
+		fmt.Fprintf(w, "--------\t-----\n")
+		fmt.Fprintf(w, "Name\t%s\n", role.Name)
+		fmt.Fprintf(w, "DisplayName\t%s\n", role.DisplayName)
+		fmt.Fprintf(w, "BuiltIn\t%v\n", role.BuiltIn)
+		fmt.Fprintf(w, "SchemeManaged\t%v\n", role.SchemeManaged)
+		for i, perm := range role.Permissions {
+			if i == 0 {
+				fmt.Fprintf(w, "Permissions\t%s\n", role.Permissions[0])
+			} else {
+				fmt.Fprintf(w, "\t%s\n", perm)
+			}
+		}
+	}
+
+	w.Flush()
+
+	return nil
+}
+
+func assignUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	role, response := c.GetRoleByName(args[0])
+	if response.Error != nil {
+		return response.Error
+	}
+
+	for _, username := range args[1:] {
+		user, response := c.GetUserByUsername(username, "")
+		if response.Error != nil {
+			return response.Error
+		}
+
+		startingRoles := strings.Fields(user.Roles)
+		for _, roleName := range startingRoles {
+			if roleName == role.Name {
+				return nil
+			}
+		}
+
+		userRoles := append(startingRoles, role.Name)
+		_, response = c.UpdateUserRoles(user.Id, strings.Join(userRoles, " "))
+		if response.Error != nil {
+			return response.Error
+		}
+	}
+
+	return nil
+}
+
+func unassignUsersCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	for _, username := range args[1:] {
+		user, response := c.GetUserByUsername(username, "")
+		if response.Error != nil {
+			return response.Error
+		}
+
+		userRoles := strings.Fields(user.Roles)
+		originalCount := len(userRoles)
+
+		for i := 0; i < len(userRoles); i++ {
+			if userRoles[i] == args[0] {
+				userRoles = append(userRoles[:i], userRoles[i+1:]...)
+				i--
+			}
+		}
+
+		if originalCount > len(userRoles) {
+			_, response = c.UpdateUserRoles(user.Id, strings.Join(userRoles, " "))
+			if response.Error != nil {
```

Comment:
Same as in the previous command, we could use `getUserFromUserArgs` here to allow for emails and ids